library/Zend/Db/Sql/AbstractSql.php

```diff
@@ -108,6 +108,13 @@ protected function processExpression(ExpressionInterface $expression, PlatformIn
         $expressionParamIndex = &$this->instanceParameterIndex[$namedParameterPrefix];
 
         foreach ($parts as $part) {
+            // #7407: use $expression->getExpression() to get the unescaped version of the expression
+            if (is_string($part) && $expression instanceof Expression) {
```

Comment:
To me It looks better then `continue` and we get rid of `throw` in a middle of a function